File: packages/shared/types/interaction.ts
/**
 * Shared Interaction types used across mobile app and API responses.
 */

export type Severity = 'critical' | 'moderate' | 'minor';
export type Mechanism = 'INHIBITS' | 'INDUCES';
export type Strength = 'strong' | 'moderate' | 'weak';

export interface InteractionChain {
  chain_id: string;
  perpetrator: string;
  perpetrator_id: string;
  victim_drug: string;
  victim_id: string;
  via_enzyme: string;
  enzyme_family: string;
  interaction_type: Mechanism;
  strengths: Strength[];
  consequence: string;
  severity: Severity;
}

export interface InteractionAlert {
  chain_id: string;
  perpetrator: string;
  victim_drug: string;
  via_enzyme: string;
  severity: Severity;
  one_line_consequence: string;
}

export interface SimulationResult {
  new_drug: string;
  affected_drug: string;
  via_enzyme: string;
  mechanism: Mechanism;
  strength: Strength;
}

File: packages/api/models/interaction.py
"""
Cascadex API — Interaction Pydantic Models.
"""

from typing import Optional

from pydantic import BaseModel


class InteractionChain(BaseModel):
    perpetrator: str
    perpetrator_id: str
    victim_drug: str
    victim_id: str
    via_enzyme: str
    enzyme_family: Optional[str] = None
    interaction_type: str
    strengths: list[str] = []
    consequence: str
    severity: Optional[str] = None


class InteractionAlert(BaseModel):
    perpetrator: str
    victim_drug: str
    via_enzyme: str
    severity: str
    consequence: str


class SimulationResult(BaseModel):
    new_drug: str
    affected_drug: str
    via_enzyme: str
    mechanism: str
    strength: Optional[str] = None
